File: html/javascripts/jquery/autorefresh.js
// -*- JavaScript -*-
//
// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
//
//                                   Jiao Lin
//                      California Institute of Technology
//                         (C) 2008 All Rights Reserved  
//
// {LicenseText}
//
// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
//



(function($) {

  $.autorefresh_start = function (timeout) {
    $.autorefresh_seconds_remained = timeout;
    $.autorefresh_update();
  };
  
  $.autorefresh_update = function () {

    if ($.autorefresh_seconds_remained==1) {
      window.location.reload();
    } else {
      $.autorefresh_seconds_remained -= 1;
      window.status=timestr($.autorefresh_seconds_remained);
      // wait 1 second to refresh
      setTimeout("$.autorefresh_update()",1000);
    }
  };

  // data
  $.autorefresh_seconds_remained = 0;

  // helpers
  function timestr(seconds) {
    curmin=Math.floor(seconds/60);
    cursec=seconds%60;
    if (curmin!=0)
      curtime=curmin+" minutes and "+cursec+" seconds left until page refresh!";
    else
      curtime=cursec+" seconds left until page refresh!";
    return curtime;
  }

 }) (jQuery);

File: vnf-alpha/vnf/content/AutoRefresh.py
#!/usr/bin/env python
#
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#
#                                  Jiao Lin
#                     California Institute of Technology
#                       (C) 2008  All Rights Reserved
#
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#


class AutoRefresh:

    def __init__(self, timeout=None):
        self.timeout = timeout
        return

    def identify(self, visitor):
        return visitor.onAutoRefresh(self)


# version
__id__ = "$Id$"

# End of file 
